From e7845d831f9c35a9b80bbf5e2b814e8227d1ebbf Mon Sep 17 00:00:00 2001
From: "Khem Raj via lists.openembedded.org"
 <raj.khem=gmail.com@lists.openembedded.org>
Date: Wed, 19 Mar 2025 18:55:29 -0700
Subject: [PATCH] opensbi: Do not inherit autotools bbclass

This package is built using simple makefiles, therefore using
autotools bbclass is not right and moreover it is now a hard
error in core

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 meta/recipes-bsp/opensbi/opensbi_1.6.bb | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/meta/recipes-bsp/opensbi/opensbi_1.6.bb b/meta/recipes-bsp/opensbi/opensbi_1.6.bb
index 3c16656b39..ed1a70d01a 100644
--- a/meta/recipes-bsp/opensbi/opensbi_1.6.bb
+++ b/meta/recipes-bsp/opensbi/opensbi_1.6.bb
@@ -6,7 +6,7 @@ LIC_FILES_CHKSUM = "file://COPYING.BSD;md5=42dd9555eb177f35150cf9aa240b61e5"
 
 require opensbi-payloads.inc
 
-inherit autotools-brokensep deploy
+inherit deploy
 
 SRCREV = "bd613dd92113f683052acfb23d9dc8ba60029e0a"
 SRC_URI = "git://github.com/riscv/opensbi.git;branch=master;protocol=https"
@@ -26,7 +26,12 @@ EXTRA_OEMAKE:append = " ${@riscv_get_extra_oemake_fdt(d)}"
 # Required if specifying a custom payload
 do_compile[depends] += "${@riscv_get_do_compile_depends(d)}"
 
-do_install:append() {
+do_compile() {
+	oe_runmake
+}
+
+do_install() {
+	oe_runmake DESTDIR=${D} install
 	# In the future these might be required as a dependency for other packages.
 	# At the moment just delete them to avoid warnings
 	rm -r ${D}/include
